src/analysis/normed_space/spectrum.lean

```diff
@@ -0,0 +1,182 @@
+
+lemma mem_resolvent_set_of_nearby {a : A} {k k' : 𝕜} (hk : k ∈ ρ a)
+  (hkk' : ∥k' - k∥ < ∥(↑hk.unit⁻¹ : A)∥⁻¹) :
+  k' ∈ ρ a :=
+begin
+  refine (units.unit_of_nearby hk.unit (↑ₐk' - a) _).is_unit,
+  calc ∥(↑ₐk' - a) - (↑ₐk - a)∥
+       = ∥↑ₐ(k' - k)∥         : by rw [ring_hom.map_sub, sub_sub_sub_cancel_right]
+  ...  = ∥k' - k∥ * ∥(1 : A)∥ : by rw [algebra.algebra_map_eq_smul_one,norm_smul]
+  ...  = ∥k' - k∥             : by simp [normed_algebra.norm_one 𝕜 A]
+  ...  < ∥↑hk.unit⁻¹∥⁻¹       : hkk',
+end
+
+lemma is_open_resolvent_set (a : A) : is_open (ρ a) :=
+begin
+  haveI := normed_algebra.nontrivial 𝕜 A,
+  apply metric.is_open_iff.mpr,
+  intros k hk,
+  refine ⟨∥↑hk.unit⁻¹∥⁻¹, inv_pos.mpr (units.norm_pos (hk.unit⁻¹)), _⟩,
+  intros k' hk',
+  rw [metric.mem_ball, dist_eq_norm] at hk',
+  exact mem_resolvent_set_of_nearby hk hk',
+end
+
+/-- The `resolvent_set` as a term of `opens 𝕜` -/
+def open_resolvent_set (a : A) : topological_space.opens 𝕜 :=
+⟨ρ a, is_open_resolvent_set a⟩
+
+lemma is_closed (a : A) : is_closed (σ a) :=
+is_open.is_closed_compl (is_open_resolvent_set a)
+
+lemma mem_resolvent_of_norm_lt {a : A} {k : 𝕜} (h : ∥a∥ < ∥k∥) :
+  k ∈ ρ a :=
+begin
+  rw [resolvent_set,set.mem_set_of_eq,algebra.algebra_map_eq_smul_one],
+  have k_pos := lt_of_le_of_lt (norm_nonneg a) h,
+  let ku := units.mk0 k (ne_zero_of_norm_pos k_pos),
+  have lt_one :=
+    calc  ∥ku⁻¹ • a∥ = ∥↑ku⁻¹ • a∥   : rfl
+      ...            = ∥(↑ku)⁻¹ • a∥ : by rw units.coe_inv' ku
+      ...            = ∥k⁻¹∥ * ∥a∥   : norm_smul k⁻¹ a
+      ...            = ∥k∥⁻¹ * ∥a∥   : by rw normed_field.norm_inv
+      ...            < 1            : (inv_mul_lt_iff k_pos).mpr (by simp [h]),
+  have : is_unit (1 - ku⁻¹ • a), from (units.one_sub (ku⁻¹ • a) lt_one).is_unit,
+  rwa ←is_unit.smul_sub_iff_sub_inv_smul at this,
+end
+
+lemma norm_le_norm_of_mem {a : A} {k : 𝕜} (hk : k ∈ σ a) :
+  ∥k∥ ≤ ∥a∥ :=
+le_of_not_lt (not_imp_not.mpr mem_resolvent_of_norm_lt hk)
+
+lemma subset_closed_ball_norm (a : A) :
+  σ a ⊆ metric.closed_ball (0 : 𝕜) (∥a∥) :=
+λ k hk, by simp [norm_le_norm_of_mem hk]
+
+lemma is_bounded (a : A) : metric.bounded (σ a) :=
+(metric.bounded_iff_subset_ball 0).mpr ⟨∥a∥,subset_closed_ball_norm a⟩
+
+theorem is_compact [proper_space 𝕜] (a : A) : is_compact (σ a) :=
+metric.is_compact_of_is_closed_bounded (is_closed a) (is_bounded a)
+
+-- this lemma is actually true without the assumption `(σ a).nonempty`,
+-- but it would require a `by_cases` split, and it's only for the slightly
+-- silly reason that `Sup (∅ : set ℝ)` is defeq `0`.
+theorem spectral_radius_le_norm (a : A) (ha : (σ a).nonempty) :
+  spectral_radius 𝕜 a ≤ ∥a∥ :=
+begin
+  have ha' : (norm '' (σ a)).nonempty, from
+    ⟨∥ha.some∥,⟨ha.some,set.nonempty.some_mem ha,rfl⟩⟩,
+  have norm_up_bd : ∥a∥ ∈ upper_bounds (norm '' (σ a)), by
+    { apply mem_upper_bounds.mpr,
+      rintros k ⟨k,⟨k_mem,k_eq⟩⟩,
+      subst k_eq,
+      exact norm_le_norm_of_mem k_mem },
```

Comment:
```suggestion
  { rintros k ⟨k, ⟨k_mem, rfl⟩⟩,
    exact norm_le_norm_of_mem k_mem },
```
`rintros` can do more work for you here.  (Note, here and elsewhere, that the style convention is to use spaces in things separated by angle brackets.)

Reply:
ah yes, someone mentioned this `rfl` syntax for `rcases` recently, I should have realized it applied to `rintros` also.